Source/Dafny/Compiler.cs

```diff
@@ -342,14 +350,25 @@ protected interface IClassWriter {
       var w = EmitAssignmentRhs(wr);
       TrExpr(rhs, w, inLetExprBody);
     }
+
     protected virtual TargetWriter EmitAssignmentRhs(TargetWriter wr) {
       wr.Write(" = ");
       var w = wr.Fork();
       EndStmt(wr);
       return w;
     }
 
-    protected virtual void EmitMultiAssignment(List<ILvalue> lhss, List<Type> lhsTypes, out List<TargetWriter> wRhss, List<Type> rhsTypes, TargetWriter wr) {
+    protected virtual string EmitAssignmentLhs(Expression e, TargetWriter wr) {
+      string target = idGenerator.FreshId("_lhs");
+      wr.Write(GenerateLhsDecl(target, e.Type, wr, null));
+      wr.Write(" = ");
+      TrExpr(e,wr,false);
+      EndStmt(wr);
```

Comment:
Use `EmitAssignment` for these 4 lines (or, if for some reason that doesn't work, at least use `EmitAssignmentRhs` for the last 3 of these lines).

Reply:
There is no signature for EmitAssignment or EmitAssignmentRhs that works here

Reply:
Does `EmitAssignmentRhs(e, false, wr);` do what L366-L368 do for all languages?